src/Tasks/AssemblyDependency/AssemblyResolution.cs

```diff
@@ -48,6 +49,7 @@ internal static class AssemblyResolution
             string sdkName,
             string rawFileNameCandidate,
             bool isPrimaryProjectReference,
+            bool isImmutableFrameworkReference,
```

Comment:
We're currently saying this is only for immutable framework references, but do you think we might extend that later? If so, maybe we should use a somewhat more generic name?

Reply:
Fair...fullyUnderstoodReferences? Better name with that intent?

Reply:
> I'd say what's interesting is not the immutability, but that we can assume its dependency closure is provided by the framework so don't need to explore it--right?

Not quite right.  The property of "dependency closure is provided" is signaled with the `ExternallyResolved` metadatum. It is a pre-existing behavior that for such assemblies we skip the dependency walk. It is happening here:
https://github.com/dotnet/msbuild/blob/768cad16718650f120dd42acfcc9cf5c4dec555f/src/Tasks/AssemblyDependency/ReferenceTable.cs#L1692-L1699

This new flag means "the file is guaranteed to exist and never change" and in the `Resolver` hierarchy it's used only in `RawFilenameResolver` to skip the file existence check.

As for the name, I was on the fence between "immutable" and "framework"  so I used both 😀